### PR TITLE
add stronger flashlight + change mp7 recoil

### PR DESCRIPTION
**Pull request by AChesheireCat**
Adds an offshoot of Carl's Flashlights to POTATO - these flashlights inherit from CUP's tactical flashlights and add a flashlight that has three modes - a beam that extends 150m but in a tiny cone, a normal light that extends like 65m in a larger cone, and a floodlight that extends 25m but in a huge cone.

Adjusts the CUP MP7 recoil to be usable - takes notes from the CUP MP5A5 recoil.

### CODE REVIEW

File: addons/strongerFlashlights/config.cpp [hunk 1]
```diff
@@ -0,0 +1,333 @@
+class CfgPatches
+{
+	class potato_stronger_flashlights
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:3]: Change to CBA/Potato standard (Everywhere)

File: addons/strongerFlashlights/config.cpp [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+	class ACE_acc_pointer_green;
+	class acc_flashlight_pistol;
+	
+/////////////////////////////////////////
+//////////Multimode Flashlight///////////
+/////////////////////////////////////////	
+	
+	class Potato_CUP_LED_flashlight_Beam: cup_acc_flashlight
+	{
+		author="Carl, Chesheire";
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:44]: Tabs -> Spaces

**Review by TheCandianVendingMachine — CHANGES_REQUESTED** [review]

**Comment by TheCandianVendingMachine** [comment]
Place for standards: https://ace3.acemod.org/wiki/development/coding-guidelines

File: addons/strongerFlashlights/$PBOPREFIX$ [hunk 3]
```diff
@@ -0,0 +1 @@
+z\potato\addons\strongerFlashLights
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/$PBOPREFIX$:1]: `strongerFlashLights` -> `strongerFlashlights` (this on me sorry)

File: addons/strongerFlashlights/CfgWeapons.hpp [hunk 4]
```diff
@@ -0,0 +1,306 @@
+class cfgWeapons
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/CfgWeapons.hpp:1]: `cfgWeapons` -> `CfgWeapons`

File: addons/strongerFlashlights/CfgWeapons.hpp [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+{
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/CfgWeapons.hpp:2]: Brace has to be on same line as `CfgWeapons`

File: addons/strongerFlashlights/CfgWeapons.hpp [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+    
+    class ItemCore;
+    class ACE_acc_pointer_green;
+    class acc_flashlight_pistol;
+    
+/////////////////////////////////////////
+//////////Multimode Flashlight///////////
+/////////////////////////////////////////    
+    
+    class GVAR(CUP_LED_flashlight_Beam): cup_acc_flashlight
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/CfgWeapons.hpp:15]: Brace has to be on same line

Same for all subsequent braces, if we have `class BLAH` we want `class BLAH {`

File: addons/strongerFlashlights/CfgWeapons.hpp [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+    {
+        author="Carl, Chesheire";
+        scope=1;
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/CfgWeapons.hpp:18]: Missing spaces between all the equals and values

File: addons/strongerFlashlights/CfgWeapons.hpp [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+        displayName="Tactical LED Flashlight (Multi-Mode)";
+        descriptionUse="Mode: Focused White Beam. Ctrl+L to change Mode";
+        descriptionShort="Mode: Focused White Beam. Ctrl+L to change Mode";
+        MRT_SwitchItemNextClass="QGVAR(CUP_LED_flashlight)";
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/CfgWeapons.hpp:22]: Incorrect usage: `MRT_SwitchItemNextClass="QGVAR(CUP_LED_flashlight)";` -> `MRT_SwitchItemNextClass = QGVAR(CUP_LED_flashlight);` `QGVAR` adds the quotes

File: addons/strongerFlashlights/script_component.hpp [hunk 9]
```diff
@@ -0,0 +1,12 @@
+#define COMPONENT strongerFlashLights
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/script_component.hpp:1]: `strongerFlashLights` -> `strongerFlashlights` (this on me, sorry)

**Review by TheCandianVendingMachine — CHANGES_REQUESTED** [review]
nearly there

File: addons/strongerFlashlights/config.cpp [hunk 10]
```diff
@@ -0,0 +1,26 @@
+class CfgPatches
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:1]: sorry chesh, one more file needs format fixes

**Review by TheCandianVendingMachine — CHANGES_REQUESTED** [review]

File: addons/strongerFlashlights/config.cpp [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+{
+	class ADDON
+	{
+		weapons[]=
+		{
+//Base
+			"GVAR(Vanilla_HG_LED_flashlight_Flood)",
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:8]: `"GVAR` -> `QGVAR

File: addons/strongerFlashlights/config.cpp (hunk 10 above)
Comment (AChesheireCat) [reply addons/strongerFlashlights/config.cpp:1]: no worries! updated :)

File: addons/strongerFlashlights/config.cpp [hunk 12]
```diff
@@ -0,0 +1,26 @@
+class CfgPatches
+{
+	class ADDON
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:3]: Braces inconsistent, need to be on same line

File: addons/strongerFlashlights/config.cpp [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		weapons[]=
+		{
+//Base
+			QGVAR(Vanilla_HG_LED_flashlight_Flood),
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:8]: Tabs -> spaces

File: addons/strongerFlashlights/config.cpp [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+			QGVAR(Vanilla_HG_LED_flashlight_Beam),
+			QGVAR(Vanilla_HG_LED_flashlight),
+//CUP flashlight
+			QGVAR(CUP_LED_flashlight_Flood),
+			QGVAR(CUP_LED_flashlight_Beam),
+			QGVAR(CUP_LED_flashlight)
+		};
+		units[]=
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:16]: `units[] = {};`

File: addons/strongerFlashlights/config.cpp [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+		{
+		};
+		requiredVersion=0.1;
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:19]: `requiredVersion=0.1` -> `requiredVersion = REQUIRED_VERSION` for consistency

File: addons/strongerFlashlights/config.cpp [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+		requiredAddons[]={"cup_weapons_east_attachments", "cup_weapons_west_attachments", "ace_laserpointer"};
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:20]: I think we want to add `skipWhenMissingDependencies = 1;`. See https://github.com/BourbonWarfare/POTATO/blob/master/addons/cswCompat/config.cpp#L8

File: addons/strongerFlashlights/config.cpp [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+		author = "Carl, Chesheire";
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:21]: `author = "Potato"`. Add `authors[] = {"Carl", "Cheshire"};`. See https://github.com/BourbonWarfare/POTATO/blob/master/addons/cswCompat/config.cpp#L10

File: addons/strongerFlashlights/config.cpp [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+		authorUrl="";
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:22]: `authorUrl = "https://github.com/BourbonWarfare/POTATO";` instead

File: addons/strongerFlashlights/config.cpp [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+	};
```
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:23]: Add `VERSION_CONFIG;`. See https://github.com/BourbonWarfare/POTATO/blob/master/addons/cswCompat/config.cpp#L12C9-L12C24

**Review by TheCandianVendingMachine — CHANGES_REQUESTED** [review]
`config.cpp` has inconsistent brace style, which were the aforementioned style issues 😅. New issues found as well :upside_down_face:

File: addons/strongerFlashlights/config.cpp (hunk 10 above)
Comment (TheCandianVendingMachine) [inline addons/strongerFlashlights/config.cpp:1]: Need to include `#include "script_component.hpp"` at top of this file

**Review by TheCandianVendingMachine — CHANGES_REQUESTED** [review]

**Review by TheCandianVendingMachine — APPROVED** [review]
lgtm, thanks for the style updates!

**Pull request merged**